### PR TITLE
Site Icon: Remove all traces

### LINKED ISSUE #3073: Site Icon: Prune old code
When WP 4.4 is released, the next major version of Jetpack will support 4.3+. This means we can remove all Site Icon code, except for any upgrade code for folks catching up to latest and greatest versions.

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes #3073

To be merged when 4.3 is minimum.  

Removes all traces of jetpack's Site Icon code.  In places like opengraph, post images, twitter cards, etc, I've changed it to use core's methods.  

It should be removed from our settings table completely and please test to make sure that your Icon is still working in Calypso as well.  

I left `jetpack_site_icon_url()` in `/modules/site-icon/site-icon-functions.php` so that we can continue checking for back-compat reasons (see `Jetpack_Sync_Module_Options()->jetpack_sync_core_icon()`)

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -6886,53 +6877,4 @@ public function dashboard_widget_connect_to_wpcom() {
...
-		if ( has_site_icon() ) {
-			if ( Jetpack::is_module_active( 'site-icon' ) ) {
-				Jetpack::log( 'deactivate', 'site-icon' );
-				Jetpack::deactivate_module( 'site-icon' );
-			}
-			return true;
-		}
-
-		// Transfer Jetpack's site icon to use core.
-		$site_icon_id = Jetpack::get_option( 'site_icon_id' );
```
Comment (kraftbj) [inline class.jetpack.php]: We should probably keep some variant of this for people who skipped updating JP for awhile.

File: modules/site-icon.php [hunk 2]
```diff
@@ -1,17 +0,0 @@
-<?php
```
Comment (kraftbj) [inline modules/site-icon.php]: I'm not sure the rationale (prevent fatals from an `include` cached?), but previously deprecated modules still have their `{$slug}.php` file present with a comment saying deprecated.

Comment (dereksmart) [inline modules/site-icon.php]: better to be safe than sorry I guess

File: class.jetpack.php (hunk 1 above)
Comment (dereksmart) [inline class.jetpack.php]: How long do we give them?  It'll be 6 months in January.

Comment (kraftbj) [inline class.jetpack.php]: I'm really conservative on upgrade pieces. WordPress itself keeps update routines in forever. My two cents is keep the upgrade pieces and burn the rest.

**Comment by kraftbj** [comment]
Now that 4.5 is out, 4.4 is the minimum and we can move on this (which has been the case for awhile, but I digress).

Per my comment above, I think we should leave the upgrade bits in for folks who are slow to update their sites, but otherwise, let's prune this bad boy.

**Comment by lancewillett** [comment]
Side note: when testing how this integrations with Calyspo, check the following cases:

1. Add new image (from nothing as default) — should appear in Calypso right away.
2. Change image — changed image should appear in Calypso right away.
3. Remove image (back to default, empty) — should disappear in Calypso sites list right away.

**Comment by lamosty** [comment]
How is the WP's native site icon synced to WP.com shadow blog when this functionality is removed? Currently, there is a `jetpack_site_icon_url` option on the WP.com shadow blog which points to the uploaded site icon on the WP blog.

**Comment by dereksmart** [comment]
@lamosty here: https://github.com/Automattic/jetpack/blob/master/sync/class.jetpack-sync-defaults.php#L40

I should be synced back and prefixed with `jepack_`

**Comment by rcoll** [comment]
@dereksmart If you would be so kind as to rebase against master, I'll promptly get this tested.

File: class.jetpack.php (hunk 1 above)
Comment (dereksmart) [inline class.jetpack.php]: @kraftbj the back compat is now handled in sync.  Check out `Jetpack_Sync_Module_Options()->jetpack_sync_core_icon()`

**Comment by dereksmart** [comment]
@rcoll rebased and ready for testing!

**Comment by samhotchkiss** [comment]
Hey @dereksmart -- one more rebase?

**Comment by dereksmart** [comment]
@samhotchkiss rebased 👍

**Pull request merged**